### PR TITLE
add `StanfordSentimentTreeBankDatasetReader.apply_token_indexers()`

### PR DESCRIPTION
**Pull request by xinzhel**
To be consistent with the latest Allennlp implementation, add `token_indexers` of `TextField` in `apply_token_indexers()` rather than `text_to_instance()`.

## Issue
When num_worker >0, the error shows up: Found a TextField (tokens) with token_indexers already applied, but you're using num_workers > 0 in your data loader. Make sure your dataset reader's text_to_instance() method doesn't add any token_indexers to the TextFields it creates. Instead, the token_indexers should be added to the instances in the apply_token_indexers() method of your dataset reader (which you'll have to implement if you haven't done so already).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   See [PR #257](https://github.com/allenai/allennlp-models/pull/257) for more details.
 - Added a `beam_search` parameter to the `generation` models so that a `BeamSearch` object can be specified in their configs.
 - Added a binary gender bias-mitigated RoBERTa model for SNLI.
+- Added `StanfordSentimentTreeBankDatasetReader.apply_token_indexers()` to add token_indexers rather than in `text_to_instance`
```
Comment (jacob-morrison) [inline CHANGELOG.md:33]: Can you move this up to line 13, under "Unreleased"?

**Review by jacob-morrison — CHANGES_REQUESTED** [review]

File: allennlp_models/classification/dataset_readers/stanford_sentiment_tree_bank.py [hunk 2]
```diff
@@ -153,3 +153,7 @@ def make_token(t: Union[str, Token]):
                     sentiment = "1"
             fields["label"] = LabelField(sentiment)
         return Instance(fields)
+
+    @overrides
+    def apply_token_indexers(self, instance: Instance) -> None:
+        instance.fields["tokens"]._token_indexers = self._token_indexers
```
Comment (epwalsh) [inline allennlp_models/classification/dataset_readers/stanford_sentiment_tree_bank.py:159]: This is equivalent but preferred:

```suggestion
        instance.fields["tokens"].token_indexers = self._token_indexers
```

**Review by epwalsh — CHANGES_REQUESTED** [review]
Thank you @xinzhel! In addition to @jacob-morrison's comment I just have one small suggestion.

File: CHANGELOG.md (hunk 1 above)
Comment (xinzhel) [reply CHANGELOG.md:33]: ok

File: allennlp_models/classification/dataset_readers/stanford_sentiment_tree_bank.py (hunk 2 above)
Comment (xinzhel) [reply allennlp_models/classification/dataset_readers/stanford_sentiment_tree_bank.py:159]: ok

**Review by epwalsh — APPROVED** [review]
LGTM!

**Review by jacob-morrison — APPROVED** [review]

**Pull request merged**